src/new/templates/rust/ui/hyperapp-skeleton/README.md

```diff
@@ -0,0 +1,251 @@
+│   ├── Cargo.toml      # Process dependencies
+│   └── src/
+│       ├── lib.rs      # Main app logic (well-commented)
+│       └── icon        # App icon file
+├── ui/                 # Frontend application
+│   ├── package.json    # Node dependencies
+│   ├── index.html      # Entry point (includes /our.js)
+│   ├── vite.config.ts  # Build configuration
+│   └── src/
+│       ├── App.tsx     # Main React component
+│       ├── store/      # Zustand state management
+│       ├── types/      # TypeScript type definitions
+│       └── utils/      # API utilities
+├── api/                # Generated WIT files (after build)
+└── pkg/                # The final build product, including manifest.json, scripts.json and built package output
+```
+
+## Key Concepts
+
+### 1. The Hyperprocess Macro
+
+The `#[hyperprocess]` macro is the core of the Hyperapp framework. It provides:
+- Async/await support without tokio
+- Automatic WIT generation
+- State persistence
+- HTTP/WebSocket endpoint configuration
+
+### 2. Required Patterns
+
+#### HTTP Endpoints
+ALL HTTP endpoints MUST be tagged with `#[http]`:
+```rust
+#[http]
+async fn my_endpoint(&self) -> String {
+    // Implementation
+}
+```
+
+#### Frontend API Calls
+Parameters must be sent as tuples for multi-parameter methods:
+```typescript
+// Single parameter
+{ "MethodName": value }
+
+// Multiple parameters
+{ "MethodName": [param1, param2] }
+```
+
+#### The /our.js Script
+MUST be included in index.html:
+```html
+<script src="/our.js"></script>
+```
+
+### 3. State Persistence
+
+Your app's state is automatically persisted based on the `save_config` option:
+- `OnDiff`: Save when state changes (strongly recommended)
+- `Never`: No automatic saves
+- `EveryMessage`: Save after each message (safest; slowest)
+- `EveeyNMessage(u64)`: Save every N messages received
+- `EveeyNSeconds(u64)`: Save every N seconds
+
+## Customization Guide
+
+### 1. Modify App State
+
+Edit `AppState` in `skeleton-app/src/lib.rs`:
+```rust
+#[derive(Default, Serialize, Deserialize)]
+pub struct AppState {
+    // Add your fields here
+    my_data: Vec<MyType>,
+}
+```
+
+### 2. Add HTTP Endpoints
+
+For UI interaction:
+```rust
+#[http]
+async fn my_method(&mut self, request_body: String) -> Result<String, String> {
+    // Parse request, update state, return response
+}
+```
+
+### 3. Add Capabilities
+
+Add system permissions in `pkg/manifest.json`:
+```json
+"request_capabilities": [
+    "homepage:homepage:sys",
+    "http-server:distro:sys",
+    "vfs:distro:sys"  // Add as needed
+]
+```
+
+These are required to message other local processes.
+They can also be granted so other local processes can message us.
+There is also a `request_networking` field that must be true to send messages over the network p2p.
+
+### 4. Update Frontend
+
+1. Add types in `ui/src/types/skeleton.ts`
+2. Add API calls in `ui/src/utils/api.ts`
+3. Update store in `ui/src/store/skeleton.ts`
+4. Modify UI in `ui/src/App.tsx`
```

Comment:
I think step 2 here should be removed; are other paths valid?

Reply:
Done. Yup, they are valid.